repo-scripts/size-analysis/analyze-all-bundles.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as tmp from 'tmp';
+
+import { Bundler, Mode, run as runBundleAnalysis } from './bundle-analysis';
+import { Report } from '../../scripts/size_report/report_binary_size';
+
+/**
+ * Runs bundle analysis for all bundle definition files under:
+ *
+ *   `firebase-js-sdk/repo-scripts/size-analysis/bundle-definitions`
+ *
+ * The method accepts an optional parameter `version`:
+ * 1. when presented (for example, `version` = '9.0.0'), this method measures the bundle size by
+ *    building test bundles with dependencies at that specific version downloaded from npm
+ * 2. when omitted (in this case, `version` = null), this method measures the bundle size by
+ *    building test bundles with dependencies from local artifacts (e.g. produced by `yarn build`)
+ *
+ * #1 is intended only for manual runs for the purpose of back-filling historical size data. #2 is
+ * intended for CI runs that measure size for the current commit.
+ *
+ * More details on how a test bundle is built can be found in `bundle-analysis.ts`.
+ *
+ * @param {string} [version] - If present, the SDK version to run measurement against
+ * @returns {Promise<Report[]>} A list of bundle size measurements
```

Comment:
We use [tsdoc](https://tsdoc.org/) instead of jsdoc. Please update. You can take a look at any of our packages for reference.

Reply:
Is that just a standard of doc format or something I should use to generate the doc?

Reply:
Done. Changed to tsdoc format.